substrate/dex/pallet/src/lib.rs

```diff
@@ -154,11 +154,34 @@ pub mod pallet {
   #[pallet::storage]
   pub type Pools<T: Config> = StorageMap<_, Blake2_128Concat, PoolId, PoolInfo<Coin>, OptionQuery>;
 
+  #[pallet::storage]
+  #[pallet::getter(fn last_quote_for_block)]
+  pub type LastQuoteForBlock<T: Config> =
+    StorageDoubleMap<_, Identity, BlockNumberFor<T>, Identity, Coin, [u8; 8], ValueQuery>;
+
+  /// Moving window of oracle prices.
+  #[pallet::storage]
+  #[pallet::getter(fn oracle_prices)]
+  pub type OraclePrices<T: Config> = StorageMap<_, Identity, [u8; 8], u16, OptionQuery>;
+  impl<T: Config> Pallet<T> {
+    /// Get the highest sustained(lowest) value of the window
```

Comment:
"""
Get the highest sustained value for this window.

This is actually the lowest price observed during the windows, as it's the price all prices are greater than or equal to.
"""

Reply:
This just has greater clarity on highest sustained == lowest.